ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/LagReportingAgentTest.java

```diff
@@ -0,0 +1,280 @@
+  private static final QueryStateStoreId QUERY_STORE_A =
+      QueryStateStoreId.of(QUERY_ID0, STATE_STORE0);
+  private static final QueryStateStoreId QUERY_STORE_B =
+      QueryStateStoreId.of(QUERY_ID1, STATE_STORE1);
+
+  private static final Map<QueryStateStoreId, Map<Integer, LagInfoEntity>> LAG_MAP1
+      = ImmutableMap.<QueryStateStoreId, Map<Integer, LagInfoEntity>>builder()
+      .put(QUERY_STORE_A, ImmutableMap.<Integer, LagInfoEntity>builder()
+          .put(1, new LagInfoEntity(1, 10, 9))
+          .put(3, new LagInfoEntity(3, 10, 7))
+          .build())
+      .put(QUERY_STORE_B, ImmutableMap.<Integer, LagInfoEntity>builder()
+          .put(4, new LagInfoEntity(6, 10, 4))
+          .build())
+      .build();
+
+  private static final Map<QueryStateStoreId, Map<Integer, LagInfoEntity>> LAG_MAP2
+      = ImmutableMap.<QueryStateStoreId, Map<Integer, LagInfoEntity>>builder()
+      .put(QUERY_STORE_A, ImmutableMap.<Integer, LagInfoEntity>builder()
+          .put(1, new LagInfoEntity(4, 10, 6))
+          .build())
+      .put(QUERY_STORE_B, ImmutableMap.<Integer, LagInfoEntity>builder()
+          .put(4, new LagInfoEntity(7, 10, 3))
+          .build())
+      .build();
+
+  @Mock
+  private PersistentQueryMetadata query0;
+  @Mock
+  private PersistentQueryMetadata query1;
+  @Mock
+  private KafkaStreams kafkaStreams0;
+  @Mock
+  private KafkaStreams kafkaStreams1;
+  @Mock
+  private ServiceContext serviceContext;
+  @Mock
+  private KsqlEngine ksqlEngine;
+  @Mock
+  private SimpleKsqlClient ksqlClient;
+  @Mock
+  private LagInfo lagInfo0;
+  @Mock
+  private LagInfo lagInfo1;
+  @Mock
+  private Clock clock;
+
+  private LagReportingAgent lagReportingAgent;
+
+
+  @Before
+  public void setUp() {
+    when(serviceContext.getKsqlClient()).thenReturn(ksqlClient);
+
+    Builder builder = LagReportingAgent.builder();
+    lagReportingAgent = builder
+        .clock(clock)
+        .build(ksqlEngine, serviceContext);
+    lagReportingAgent.setLocalAddress(LOCALHOST_URL);
+  }
+
+  @Test
+  public void shouldReceiveLags() {
+    // When:
+    lagReportingAgent.receiveHostLag(hostLag(HOST1, LAG_MAP1, 100));
+    lagReportingAgent.receiveHostLag(hostLag(HOST2, LAG_MAP2, 200));
+    lagReportingAgent.onHostStatusUpdated(HOSTS_ALIVE);
+
+    // Then:
+    Map<HostInfo, LagInfoEntity> hostPartitionLagList
+        = lagReportingAgent.getHostsPartitionLagInfo(HOSTS, QueryStateStoreId.of("query0", "a"), 1);
+    assertEquals(2, hostPartitionLagList.size());
+    assertEquals(1, hostPartitionLagList.get(HI1).getCurrentOffsetPosition());
+    assertEquals(10, hostPartitionLagList.get(HI1).getEndOffsetPosition());
+    assertEquals(9, hostPartitionLagList.get(HI1).getOffsetLag());
+    assertEquals(4, hostPartitionLagList.get(HI2).getCurrentOffsetPosition());
+    assertEquals(10, hostPartitionLagList.get(HI2).getEndOffsetPosition());
+    assertEquals(6, hostPartitionLagList.get(HI2).getOffsetLag());
```

Comment:
nit: this might be a pain, but I think these tests would be easier to read and maintain if we had properly named constants for all of these ints